test/integration/client-side-operations-timeout/client_side_operations_timeout.unit.test.ts

```diff
@@ -195,11 +196,10 @@ describe('CSOT spec unit tests', function () {
               {
                 autoEncryption: {
                   extraOptions: {
-                    mongocryptdBypassSpawn: true,
-                    mongocryptdURI: 'mongodb://localhost:27017/db?serverSelectionTimeoutMS=1000',
+                    mongocryptdURI: 'mongodb://localhost:27020/db?serverSelectionTimeoutMS=2000',
```

Comment:
This was an odd bug I found.  I'm not sure why we weren't seeing this issue before, but with the changes in this PR, the test was failing with a "You must set server API" server selection error on our server API tasks.

Turns out the issue was caused by two things together:

1. the cryptd URL happens to be the same as the server, which means that we attempted to connect to the test server, not a mongocryptd
2. mongocryptd clients are not created with configuration.newClient() - so the mongocryptd client did not have the server API version set.